libraries/YarpPlugins/AravisGigE/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+# Copyright: (C) 2017 Universidad Carlos III de Madrid
+# Authors: Juan G. Victores & David Estevez
+# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+
+cmake_minimum_required(VERSION 2.6)
+
+find_package(YARP REQUIRED)
+include_directories(${YARP_INCLUDE_DIRS}) # deprecated
+list(APPEND CMAKE_MODULE_PATH ${YARP_MODULE_PATH}) # deprecated
+include(YarpPlugin) # deprecated
+
+YARP_PREPARE_PLUGIN(AravisGigE CATEGORY device TYPE roboticslab::AravisGigE INCLUDE AravisGigE.hpp WRAPPER grabber)
+
+IF (NOT SKIP_AravisGigE)
+  find_package(ARAVIS REQUIRED)
+  include_directories(${ARAVIS_INCLUDE_DIRS})
+
+  # Include any directories needed for YARP
+  include_directories(${YARP_INCLUDE_DIRS} ${ROBOTICSLAB_YARP_DEVICES_INCLUDE_DIRS} ${COLOR_DEBUG_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR})
+
+  yarp_add_plugin(AravisGigE AravisGigE.hpp DeviceDriverImpl.cpp IFrameGrabberImpl.cpp IFrameGrabberImageRawImpl.cpp IFrameGrabberControls2Impl.cpp)
+  TARGET_COMPILE_FEATURES(AravisGigE PRIVATE cxx_auto_type)
```

Comment:
Requires CMake 3.1: https://cmake.org/cmake/help/v3.9/release/3.1.html#commands.

Reply:
Do you guys intend to upgrade CMake soon or should I use the current version?

Reply:
I think that it doesn't matter for now as it's just another optional component in this repo (I added a few more devices recently that require downloading SDKs and such, thus those are clearly out of scope for ancient Debian 6.0 environments). What I would like to stress here is that `CMAKE_MINIMUM_REQUIRED` does not reflect the current requirements for this module, hence portability is compromised.

Along these lines, and as a note for future development, we should really start providing a convenient version specifier in `find_package` (for instance, code that uses recent YARP features should force a CMake configuration failure in unsupported systems - *before* even reaching the compilation step).

Reply:
Even though I like the simplicity of the auto keyword, I prefer not to add any additional dependencies to my device, specially if it could lead to confusion since some devices require one version of CMake and others require a different one. Therefor I have changed the few auto keywords I used by the corresponding iterator.

Reply:
Dispite the fact that Travis CI broken (which I've documented at #134), it looks good to me.

@David-Estevez Thanks, I'm happy with fc77463db23964622e9c5aa4de049149474bf491

@PeterBowman Okay to merge?

Reply:
@jgvictores 👍